src/Data/Repositories/Interfaces/IRepository.cs

```diff
@@ -49,7 +49,7 @@ public interface IRepository<T>
         /// </summary>
         /// <param name="T"></param>
         /// <returns>A tuple (bool, string). The bool represents the call success and the string any possible message.</returns>
-        public Task<(bool, string?)> AddRangeAsync(List<T> T, ApplicationDbContext applicationDbContext);
+        public Task<(bool, string?)> AddRangeAsync<T>(List<T> entities, ApplicationDbContext applicationDbContext) where T : class;
```

Comment:
nit. should change the generic name: 
```
Type parameter 'T' has the same name as the type parameter from outer type 'IRepository<T>'
```

Reply:
Done 🏁

Reply:
This produces a change to add to all the calls to this method. This change adds a lot of noise to the PR IMO. We can do it later.